### ISSUE TITLE
German upper umlaut not working

### ISSUE THREAD

**Issue by frankbenoit** [sunaku/glove80-keymaps#5]
Hi

on windows, for the world layer, the diaeresis symbol are not completely working.

a, e, i, o, u
in lower case, all working
in upper case only Ï, Ë are working. Ä, Ö, Ü are not generating output for me.

Is this a bug in the keymap or something wrong in my setup?

Frank

**Comment by sunaku** [comment]
Oh interesting, I wonder if this is caused by the native Alt-keycodes in the "localizing" section of the `world.yaml` file. 🤔  We can try [commenting out those characters](https://github.com/sunaku/glove80-keymaps/blob/v34/world.yaml#L124-L127) and rebuilding -- or better yet, type the Alt-keycodes manually in Windows to check if they produce the correct characters.

**Comment by sunaku** [comment]
I was able to reproduce the issue now on Windows 10 :+1: and it appears to be a matter of timing: I could manually input the ALT+0196 sequence (tapping each numpad digit slowly) to produce Ä successfully, but the same thing doesn't work when executed (very quickly) from the World layer's ZMK macro. :thinking:

**Comment by sunaku** [comment]
I was wrong: it isn't a timing issue but rather a modifier stacking issue on nested mod-morphs.  As a workaround, I've rebuilt this keymap *without* the OS-native `localizing` section for you:
- Snapshot: https://my.glove80.com/#/layout/user/25f441b0-2499-4d76-8d7c-ec58f96f90bc

Please try this out and let me know if it fixes the issue for you (using WinCompose I assume?).

**Comment by stefango79** [comment]
Awesome work. Here's what I had to do to make it work for üÜäÄöÖ on Windows 10 (nothing major, but may help others):
- Install WinCompose (didn't need it so far and not allowed on company notebook)
- Switch back to German input in Windows
- Add `#define OPERATING_SYSTEM 'W'` in the keymap (left it on EN_US)

I wanted to help here as I had the same issue in the beginning. I ended up using EN US International input which gives me the AltGr combos. Have a look here: https://de.wikipedia.org/wiki/Tastaturbelegung_US-International

**Comment by frankbenoit** [comment]
Hi sorry for the late reply...

yes I have WinCompose running

@sunaku yes, with the fix the upper umlauts are working on windows for me.

Is the "sharp S" ß , missing? Essential in German.

The key with the dollar, the euro variant works, but the other variations for pound+yen+.. are not working for me.

The world layer activation key is hard for me to type because it is mechanically not meant for the thumb, especially then in combination with the lctl and shift.
My solution - as a 80% english, 20% german typer - to have alt-gr combos with the support of wincompose

````
<Multi_key> <a> : "ä"
<Multi_key> <A> : "'Ä"
...
<Multi_key> <s> : "'ß"
````

with those I am able to access German letters on the base layer. E.g. hold 'w' (engrammer alt-gr) plus stroke the 'a' gives the 'ä'.

@stefango79 the international comes with the drawback of extra strokes for single/double quotes.

**Comment by sunaku** [comment]
I've been trying different ideas to solve issue (caused by the presence of the Ctrl key) but nothing has been completely successful so far. :thinking: This may take some time, hang in there!

The World layer is more like a proof of concept that I've put together to demonstrate a possible arrangement of international characters, so you needn't be constrained by it for actual usage.  You can take any of the ZMK macros from the World layer and put them on any other layer you wish, including the base layer.  Moreover, you can take the underlying forms:
- `&world_a_base` is the top-level construct (built from individual forms) on the World layer
- `&world_a_diaeresis` is the medium-level construct, built from lowercase and uppercase letters
- `&world_a_diaeresis_lower` is the lowest-level construct, representing the lowercase ä letter
- `&world_a_diaeresis_upper` is the lowest-level construct, representing the lowercase Ä letter

Any of those ZMK macros can be assigned directly to any key on any layer in the Glove80 Layout Editor.

**Comment by sunaku** [comment]
I've fixed the problem of Ctrl interfering with Windows Alt-keycodes in commit b97c44aae0d05ba929731043f36bf4a260210c6e, but this has introduced an additional 300ms wait before the Alt-keycode ZMK macro can be executed.  Please try this snapshot and let me know how it fares for you:
- https://my.glove80.com/#/layout/user/16d15131-6d4e-463a-be2a-3a2b6923c6e9

Per my observations, WinCompose is far more performant and reliable than Windows Alt-keycodes so I would encourage you to try commenting out the `#define WORLD_USE_COMPOSE` setting so that WinCompose is triggered by Unicode macros instead. :trophy:

**Comment by sunaku** [comment]
I've finally solved this nested mod-morph problem with another tip from @urob, which I didn't understand at first :sweat_smile: but ended up arriving at the same :star_struck: after working out the mechanics: a mod-morph likes to reuse incoming mods, so when I have a Linux Compose sequence (using RALT) being invoked with an incoming RALT, I need to preserve it (and mask out everything else). :police_officer:

@frankbenoit - please try the following snapshot on Windows for the ä/Ä key on World layer.

- Keymap: https://my.glove80.com/#/layout/user/906f984a-8e8a-453a-a044-b9749af5264b
- Commit: 87b99d7f043d405ec7e04fe9b6ba1ed98a568e98

**Comment by sunaku** [comment]
Fixed and verified as of commit 498d22b08f3f3dd678515f3f72f085fd201374be. ✅
